🐛[BUG]: ECMWF IFS Tests failing

### Version

main

### On which installation method(s) does this occur?

_No response_

### Describe the issue

FAILED test/data/test_ecmwf.py::test_ifs_time_available[msl-time1-IFS_ENS] - Failed: Timeout (>30.0s) from pytest-timeout.
FAILED test/data/test_ecmwf.py::test_ifs_time_available[msl-time1-IFS] - Failed: Timeout (>30.0s) from pytest-timeout.

Not working, and breaking CI.

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -30,6 +30,9 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ### Fixed
 
+- Fixed NNJA observation sources blocking the shared fsspec IO loop with
+  CPU-bound PrepBUFR decode work, which stalled concurrent fetches from other
+  data sources.
 - Fixed ACE2 distributed inference failures caused by nondeterministic variable ordering
   across MPI ranks.
 - Improved ACE2ERA5 inference performance by caching yearly forcing values

--- a/earth2studio/data/nnja.py
+++ b/earth2studio/data/nnja.py
@@ -22,6 +22,7 @@
 
 from __future__ import annotations
 
+import asyncio
 import hashlib
 import os
 import pathlib
@@ -212,7 +213,12 @@ class _NNJAObsBase:
                 verbose=(not self._verbose),
             )
 
-        df = self._compile_dataframe(async_tasks, variable_list, schema)
+        # Decoding is CPU-bound and internally blocks on pool futures; run it in
+        # a worker thread so it does not stall the shared fsspec IO loop that
+        # _sync_async dispatches every data source's fetch onto.
+        df = await asyncio.to_thread(
+            self._compile_dataframe, async_tasks, variable_list, schema
+        )
         df.attrs["source"] = self.SOURCE_ID
         return df
 

